vite.config.ts

```diff
@@ -2,14 +2,28 @@ import { defineConfig } from 'vite'
 import react from '@vitejs/plugin-react'
 
 // https://vitejs.dev/config/
-export default defineConfig({
+export default defineConfig(({ command }) => ({
   plugins: [react()],
+  root: 'src',
+  base: command === 'serve' ? '' : '/dist/',
+  build: {
+    // output dir for production build
+    outDir: '../dist',
+    emptyOutDir: true,
+    // emit manifest so PHP can find the hashed files
+    manifest: true,
```

Comment:
We have to do some configuration around `build` to work with PHP world.

We choose the output directory (this might need to be different in a wordpress context)

We also generate a manifest that we can read from.